course/auth.py

```diff
@@ -120,13 +122,23 @@ def get_impersonable_user_qset(impersonator):
     return impersonable_user_qset
 
 
+def _get_current_course_from_request(request):
+    course_match = re.match("^/course/"+COURSE_ID_REGEX+"/", request.get_full_path())
+    if course_match is None:
+        return None
+    else:
+        return course_match.group("course_identifier")
+
+
 class ImpersonateMiddleware(object):
     def __init__(self, get_response):
         self.get_response = get_response
 
     def __call__(self, request):
         if 'impersonate_id' in request.session:
             imp_id = request.session['impersonate_id']
+            imp_course = request.session['impersonate_course']
```

Comment:
Could you change this variable to be end in `_id` (assuming you take the other suggestion on what goes in the session)?